### DISCUSSION TITLE
Open a set of URLs in a specific Chrome profile

### DISCUSSION THREAD

**Discussion by bw352** [ttscoff/bunch-docs#232] [Q&A] [answered]
Following this syntax, how can I specify these pages open in a specific Chrome profile?

Google Chrome
- https://website.com
- https://website.com
- https://website.com

**Comment by ttscoff** [comment]
Unless you know of a way to switch profiles using scripting, I don't believe you can control this.

**Reply by roelvangils** [reply]
I would love to see support for Google Chrome profiles beging added. Switching between profiles and opening URLs in separate profiles is an integral part of my workflow.

I see that [Velja.app](https://sindresorhus.com/velja) has good support for it.

![20220724_144727_System Preferences@2x](https://user-images.githubusercontent.com/103744/180648580-f2735cc8-3dce-4128-b6ca-ad2686dbab9d.png)

![20220724_144849_Safari@2x](https://user-images.githubusercontent.com/103744/180648607-244042dc-f76f-4c76-9c65-276fa630b054.png)

Maybe [@sindresorhus](https://github.com/sindresorhus) can help?

**Comment by ttscoff** [comment]
You could, of course, send the command above as a shell command, 
substituting the URL for {URL} and adding the profile ID...

**Comment by ttscoff** [comment, marked as answer]
I got up too early and decided to look into this further. As of beta 150, you can now specify a profile (by name) when using the `chrome:` prefix, e.g.:

```
chrome[Work]:https://brettterpstra.com
```

It also works with `canary:` (which I just added as a browser option as well).

**Reply by roelvangils** [reply]
That is really cool. Thanks!

**Comment by ttscoff** [comment]
It looks like doing this would require the user to look up the profile 
ID, and then require me to call the Chrome binary directly as opposed to 
using the `open -a` command. Doable, but would take some work and I'm 
unsure the demand is significant enough to justify it.

/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome {URL} 
--profile-directory={profile_ID}

-Brett
